### PR TITLE
Major WI performance improvements on UI usage

### PR DESCRIPTION
**Pull request by Wolfsblvt**
@Succubyss' fault I have wasted time on this. Hopefully some people will get some use out of it.
This likely makes #2607 obsolete and replaces the idea behind it. @steve02081504 would be happy if you check it out and see if it works like what you wanted to achieve. Sorry for taking this in my hands and *going over you*.

This PR does three things. Two major improvements, one minor. I'll try to explain those, and also show performance logs as an example.

## 1. Performance on World Info Initial Load

> TLDR: Roughly halved the initial load time

So, the biggest issue with initial load is that when calling `showWorldEditor`, the WI entries list gets initialized, but for *some reason* (which I now know), the actual value initialization happend after.

Thing is, the `getWorldEntry` inside the pagination loading callback was synchronous. So it seems like it executed everything it could to sync at once, but all the input/change event handlers that were defined as async, did execute after the pagination was set up. And every single of those 10+ controls for all the entries needed a style recalculation. Which really, really slowed the process down.

The fix was so easy, that I am bonking myself against the head I didn't find it earlier.  
Seems like if you make the whole function async, triggering the input/change events can correctly run and be awaited **inside** this function, so all during the pagination rendering process, no need for multiple style recalculations.
You can see the results below.

*(Following screenshots are from a very normal lorebook with 140 entries, page size set to 50)*

### Staging Branch
~2.0s - rendering the pagination, then "updating" all the values inside the entries after
image

### This PR
~1.0s - 800ms spend on rendering the pagination, then 200ms on calculating style and actually showing the list
image

Everything neatly gets handled inside the `displayWorldEntries`, and then after the style rendering and actual pagination show happens.

## 2. Performance on Expand/Close All

> TLDR: From delayed to nearly instant, snappy

Expanding or closing all entries was bound to a button, that had a very simple code that just triggered the expand/close on each individual entry shown. That ran the jQuery animations behind, which for longer lists (10 or more) was both very, very slow and also CPU-intensive, as it had to recalculate all the styles.

Generally, I figure if you open/close all entries, this animation isn't really needed anyway. It *looks cool* if you simple toggle one entry, but it's kinda useless for the expand/close all.
Because the general drawer handling is all jQuery and it's difficult to hook into there, I made a new helper function that does the same, opening or closing a drawer, updating the icon, simple with JS and CSS, without jQuery or animations.

*(Following screenshots are from a very normal lorebook with 140 entries, page size set to 50)*

### Staging Branch
~1.8s - Expanding all entries, you can see the purple style recalc blocks for each individual event
image

### This PR
~ 0.3s - Hard to measure, as the event ends before the actual browser can "display" the entries. Not sure if its just too fast, lol
image

## 3. Performance on Fancy Key Edit (select2 control entries)

> TLDR: Most cases not noticeable, but huuuuuuuge performance improvements for entries with many keys

This is much, much smaller compared to the others. In most general and common scenarios that users might experience.
This is the biggest of the three changes here though in case of **actual** improvements, if you look at it with artificial tests.

If you have "fancy" edit mode chosen for the keys, select2 controls are initialized and rendered for each entry. Until now, the control was initialized and rendered, and then the existing keys were added to the `<option>` element one by one, triggering the change event and all.

The issue was the `enableKeysInput`, and especially the `select2ModifyOptions` call that added, initialized and selected the options after.

This change does two major things here:
- Initialize and select options **before** the select2 is initialized. This is just fancypants selector anyway, the normal input handles the options good enough.
- Adding all existing keys as options collects both the options and the new-to-select ones and adds them at once.

*(Following artificial test case has a lorebook with 25 entries, page size 25, and each entry has 100 primary keys listed)*

### Staging Branch
~ 38s - Just loading the list, incredibly slow. Highlighted is the `enableKeysInput` block with the select2 options.
image

### This PR
~ 1.5s - Same list. Significant time of the load is the again highlighted `enableKeysInput`, but this time it's *only* caused by the initial select2 init
image

With that artificial test, that one looks insane, doesn't it?

#### Also additional small fix:
- [x] make drag handle of WI entries on custom sorting not selectable.
  Maybe this fixes underscore's issue of it *feeling wrong*

## Checklist:

- [x] I have read the [Contributing guidelines](https://www.youtube.com/watch?v=dQw4w9WgXcQ).

### CODE REVIEW

**Review by Cohee1207 — APPROVED** [review]
Indeedy, it's very fast now.

File: public/scripts/world-info.js [hunk 1]
```diff
@@ -1867,7 +1867,8 @@ function displayWorldEntries(name, data, navigation = navigation_option.none, fl
                     Trigger %
                 </small>
             </div>`;
-            const blocks = page.map(entry => getWorldEntry(name, data, entry)).filter(x => x);
+            const blocksPromises = page.map(async (entry) => await getWorldEntry(name, data, entry)).filter(x => x);
```
Comment (Cohee1207) [inline public/scripts/world-info.js:1878]: Funnily enough, I reverted a similar change several PRs ago

Comment (Wolfsblvt) [reply public/scripts/world-info.js:1878]: Do you remember why?

Comment (Cohee1207) [reply public/scripts/world-info.js:1878]: It didn't make sense to make it async without actually awaiting any operations inside.

Comment (Wolfsblvt) [reply public/scripts/world-info.js:1878]: Well, we now learned that the JS engine sees that differently.

My limited knowledge would guess: If you trigger an event with `.trigger('change')`, and the defined event handler is async, it'll implicitly execute this **after**, without being able to do that in the same execution/render step.
